etl/steps/data/garden/energy/2022-12-12/electricity_mix.py

```diff
@@ -0,0 +1,324 @@
+        ],
+        "low_carbon_generation__twh": [
+            "renewable_generation__twh",
+            "nuclear_generation__twh",
+        ],
+    }
+
+    # Create new columns, by adding up other columns (and allowing for only one nan in each sum).
+    for new_column in aggregates:
+        table_bp[new_column] = pd.DataFrame(table_bp)[aggregates[new_column]].sum(
+            axis=1, min_count=len(aggregates[new_column]) - 1
+        )
+
+    # Prepare data in a dataframe with a dummy index.
+    df_bp = pd.DataFrame(table_bp).reset_index()
+
+    return df_bp
+
+
+def process_ember_data(table_ember: catalog.Table) -> pd.DataFrame:
+    """Load necessary columns from the Combined Electricity Review and prepare a dataframe with the required variables.
+
+    Parameters
+    ----------
+    table_ember : catalog.Table
+        Combined Electricity Review (combination of Ember's Global and European Electricity Reviews).
+
+    Returns
+    -------
+    df_ember : pd.DataFrame
+        Processed Combined Electricity Review.
+
+    """
+    # Columns to load from Ember dataset.
+    columns = {
+        "generation__bioenergy__twh": "bioenergy_generation__twh",
+        "generation__gas__twh": "gas_generation__twh",
+        "generation__coal__twh": "coal_generation__twh",
+        "generation__other_fossil__twh": "oil_generation__twh",
+        "generation__renewables__twh": "renewable_generation__twh",
+        "generation__other_renewables__twh": "other_renewables_excluding_bioenergy_generation__twh",
+        "generation__clean__twh": "low_carbon_generation__twh",
+        "generation__hydro__twh": "hydro_generation__twh",
+        "generation__nuclear__twh": "nuclear_generation__twh",
+        "generation__solar__twh": "solar_generation__twh",
+        "generation__wind__twh": "wind_generation__twh",
+        "generation__fossil__twh": "fossil_generation__twh",
+        "generation__total_generation__twh": "total_generation__twh",
+        "demand__total_demand__twh": "total_demand__twh",
+        "emissions__total_emissions__mtco2": "total_emissions__mtco2",
+        "emissions__co2_intensity__gco2_kwh": "co2_intensity__gco2_kwh",
+        "imports__total_net_imports__twh": "total_net_imports__twh",
+    }
+    table_ember = table_ember[list(columns)].rename(columns=columns, errors="raise")
+
+    # In BP data, there is a variable "Geo Biomass Other", which combines all other renewables.
+    # In Ember data, "other rewenables" excludes bioenergy.
+    # To be able to combine both datasets, create a new variable for generation of other renewables including bioenergy.
+    table_ember["other_renewables_including_bioenergy_generation__twh"] = (
+        pd.DataFrame(table_ember)["other_renewables_excluding_bioenergy_generation__twh"]
+        + table_ember["bioenergy_generation__twh"]
+    )
+
+    # Prepare data in a dataframe with a dummy index.
+    df_ember = pd.DataFrame(table_ember).reset_index()
```

Comment:
Idem to my previous two comments for `table_bp`. Perhaps worth defining `df_ember = pd.DataFrame(table_ember)`.

Just for readability purposes.